### ISSUE TITLE
Support using along terraform-google-provider 4.x.x

### ISSUE THREAD

**Issue by jackwhelpton** [terraform-google-modules/terraform-google-bastion-host#97]
### TL;DR

Need to be able to use the Google provider v4, required by upstream modules.

### Expected behavior

Module is usable with Google provider 4.0.

### Observed behavior

Could not retrieve the list of available versions for provider hashicorp/google: no available releases match the
│ given constraints >= 2.12.0, >= 3.43.0, >= 3.45.0, ~> 3.53, >= 4.0.0, < 4.0.0, < 5.0.0

### Terraform Configuration

```hcl
n/a
```

### Terraform Version

```sh
Terraform v1.0.11
on windows_amd64
+ provider registry.terraform.io/hashicorp/external v2.1.0
+ provider registry.terraform.io/hashicorp/google v3.90.0
+ provider registry.terraform.io/hashicorp/google-beta v3.90.0
+ provider registry.terraform.io/hashicorp/helm v2.3.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.6.1
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/netapp/netapp-gcp v21.5.0
```

### Additional information

The `terraform-google-vm` module has already been updated: that is one of the dependencies that currently pins the version here. Until that's released, I'll use the `master` branch from that repo to unblock myself while fixing everything else.

**Comment by jackwhelpton** [comment]
PR (work in progress): https://github.com/terraform-google-modules/terraform-google-bastion-host/pull/98

**Comment by bharathkkb** [comment]
fixed by https://github.com/terraform-google-modules/terraform-google-bastion-host/pull/98
